src/components/fieldset/Fieldset.js

```diff
@@ -40,4 +41,9 @@ export default class FieldsetComponent extends NestedComponent {
     super(...args);
     this.noField = true;
   }
+
+  createComponent(component, options, data, before) {
+    component.id =  getRandomComponentId();
+    return super.createComponent(component, options, data, before);
```

Comment:
I have checked the ticket. Does this issue effect only Fieldset component? I am pretty sure that all nested components (like Panel, Container and other) inside NestedArray components have the same issue with ids. It worth moving the fix to the NestedComponent/NestedDataComponent/NestedArrayComponent level.

Reply:
added the fix tio the NestedComponent